packages/worker/src/worker.ts

```diff
@@ -1630,6 +1710,16 @@ export class Worker {
               })
             )
           ).pipe(
+            // Reinject fatalError inside the stream, if any,
+            // with precedence over the error from the pipe
```

Comment:
A bit more info on the "why" could be useful here...
RxJS is just complicating things here...